spec/index.md

```diff
@@ -234,6 +236,8 @@ The additional features provided by the Interface (KL27) via I2C are:
     - The main (nRF) must wait for the `COMBINED_SENSOR_INT` signal to be asserted by the secondary (KL27)
 - `write_request` can be sent by both secondary and main.
     - For the secondary to initiate this, it must assert the interrupt signal first and then the main must poll (i2c read) the device for data.
+- I2C transactions must not overlap. Every I2C Write, must be followed by an I2C Read. 
```

Comment:
> Every I2C Write, must be followed by an I2C Read.

Does this affect a read command from CODAL that triggered by a different device (motion sensors) in the shared interrupt signal ?

Reply:
Not sure I quite understand that @microbit-carlos. Transactions are serialised, so should be ok?

There may be another I2C transaction on the bus (say to the motion sensor) while we wait for a response from the KL27, 
but that should be ok right?

Reply:
What kind of I2C transaction does CODAL do when the shared interrupt is triggered?
Is it and I2C write, followed by an I2C read?

Reply:
The I2C write would occur before the IRQ. So on an IRQ, we just read:

https://github.com/lancaster-university/codal-microbit-v2/blob/master/source/MicroBitUSBFlashManager.cpp#L579
https://github.com/lancaster-university/codal-microbit-v2/blob/master/source/MicroBitPowerManager.cpp#L341

Reply:
Right, yeah, that is not a problem, since any previous "I2C write" started by CODAL should have had their respective "I2C read" before the IRQ was serviced.

I guess my only concern is that we are basically saying all "I2C write" transactions require an "I2C read".
I assume there are I2C write commands that do not require an I2C read, right?

Reply:
I guess so... but we always read the result to get a return code.  :)